website/blog/2023-02-21-ferretdb-compare-arrays.md

```diff
@@ -0,0 +1,380 @@
+
+### `$gt` operator comparison
+
+[$gt](https://docs.ferretdb.io/basic_operations/read/#retrieve-documents-using-operator-queries)
+filter operators select records that are greater than the query.
+
+We use the `gtexample` collection.
+
+```js
+db.gtexample.insertMany([
+  { _id: 'scalar-null', v: null },
+  { _id: 'scalar-number', v: 1 },
+  { _id: 'scalar-string', v: 'a' },
+  { _id: 'array-empty', v: [] },
+  { _id: 'array-null', v: [ null ] },
+  { _id: 'array-mixed', v: [ 1, 'z', 'b' ] },
+  { _id: 'array-string', v: [ 'd' ] },
+  { _id: 'array-bool', v: [ false ] },
+  { _id: 'nested-empty', v: [ [] ] },
+  { _id: 'nested-mixed', v: [ [ 'c' ], [ 1, 'z', 'b' ] ] },
+  { _id: 'nested-string', v: [ [ 'a' ] ] },
+  { _id: 'nested-bool', v: [ [ true ] ] }
+])
+```
+
+**Example:** `$gt` Operator with Scalar Argument
+
+The following uses `$gt` to select records that are greater than `'c'`.
+
+```js
+db.gtexample.find({v: {$gt: 'c'}})
+[
+  { _id: 'array-mixed', v: [ 1, 'z', 'b' ] },
+  { _id: 'array-string', v: [ 'd' ] }
+]
+```
+
+The records selected are `array-mixed` and `array-string`.
+They are both arrays contain string BSON value that is greater than the filter argument of `c`.
+
+It did not select arrays that does not contain string BSON type such as `array-null`.
+
+No nested arrays were selected because array is not the same BSON type as string.
+
+**Example:** `$gt` Operator with Array Argument
+
+The following demonstrates the array filter argument `['c']` using the `$gt` operator.
+
+In the previous section, `$lt` array argument operator selected arrays that are less
+and nested arrays that contain less.
+
+:::caution
+This example shows an uncertain result of `$gt` operator.
+:::
+
+```js
+db.gtexample.find({v: {$gt: ['c']}})
+```
+
+The output:
+
+```js
+[
+  { _id: 'array-string', v: [ 'd' ] },
+  { _id: 'array-bool', v: [ false ] },
+  { _id: 'nested-empty', v: [ [] ] },
+  { _id: 'nested-mixed', v: [ [ 'c' ], [ 1, 'z', 'b' ] ] },
+  { _id: 'nested-string', v: [ [ 'g', 'z' ] ] },
+  { _id: 'nested-bool', v: [ [ true ] ] }
+]
+```
+
+No scalar documents were selected.
+
+Here, we witness the same behaviour of `$lt` on array argument because it only selects array BSON type.
+
+In the query, `array-bool` and `array-string` were selected since they contain values that is
+greater than the filter argument of `c` at its first index.
+
+The arrays are compared index to index using BSON comparison order then by value comparison.
+
+When dealing with nested arrays, using the `$gt` operator on an array filter argument does not select the largest element from a nested array, unlike how `$lt` operates.
+
+Instead, the `$gt` operator compares the array filter argument and the nested array in the collection index to index, using the BSON comparison order.
+
+If the BSON type of the array in the collection is greater than the BSON type of the filter argument (which is a string), then all nested arrays are selected.
```

Comment:
```suggestion
Because BSON type of the array in the collection is greater than the BSON type of the filter argument (which is a string), all nested arrays are selected.
```

This was my reasoning due to my finding